pumps/syslog.go

```diff
@@ -0,0 +1,140 @@
+package pumps
+
+import (
+	"fmt"
+	"github.com/TykTechnologies/logrus"
+	"github.com/TykTechnologies/tyk-pump/analytics"
+	"github.com/mitchellh/mapstructure"
+	"log/syslog"
+)
+
+type SyslogPump struct {
+	syslogConf *SyslogConf
+	Writer     *syslog.Writer
+}
+
+var (
+	logPrefix = "syslog-pump"
+)
+
+type SyslogConf struct {
+	Transport   string `mapstructure:"transport"`
+	NetworkAddr string `mapstructure:"network_addr"`
+	LogLevel    int    `mapstructure:"log_level"`
+}
+
+func (s *SyslogPump) GetName() string {
+	return "Syslog Pump"
+}
+
+func (s *SyslogPump) New() Pump {
+	newPump := SyslogPump{}
+	return &newPump
+}
+
+func (s *SyslogPump) Init(config interface{}) error {
+	//Read configuration file
+	s.syslogConf = &SyslogConf{}
+	err := mapstructure.Decode(config, &s.syslogConf)
+
+	if err != nil {
+		log.WithFields(logrus.Fields{
+			"prefix": logPrefix,
+		}).Fatal("Failed to decode configuration: ", err)
```

Comment:
Should we `return` here?

Reply:
We don't need to. `Fatal` will shut down the process

Reply:
I prefer to shut down the pump if the desired pumps aren't loading correctly